functorch/functorch/_src/aot_autograd.py

```diff
@@ -174,146 +175,192 @@ def call_func_with_args(f, args, steal_args=False):
+    with torch.no_grad():
+        fw_outs = call_func_with_args(compiled_fw, flat_args)
 
-                with track_graph_compiling("forward"):
-                    compiled_fw = fw_compiler(fw_module, flat_tensor_args)
+    if config.debug_partitioner:
+        activation_sizes = 0
+        for out in fw_outs[num_outs:]:
+            if isinstance(out, torch.Tensor):
+                activation_sizes += out.storage().nbytes()
+        print(f"Real Activations Stored(GB): {activation_sizes/1e9}")
 
-                fw_outs = call_func_with_args(compiled_fw, flat_tensor_args)
-                if config.debug_partitioner:
-                    activation_sizes = 0
-                    for out in fw_outs[num_outs:]:
-                        if isinstance(out, torch.Tensor):
-                            activation_sizes += out.storage().nbytes()
-                    print(f"Real Activations Stored(GB): {activation_sizes/1e9}")
+    bw_args = fw_outs[num_outs:] + fw_outs[0:num_outs]
+    with track_graph_compiling("backward", True):
+        compiled_bw = aot_config.bw_compiler(bw_module, bw_args)
+    torch._C._jit_set_autocast_mode(old_jit_autocast_flag)
 
-                bw_args = fw_outs[num_outs:] + fw_outs[0:num_outs]
-                with track_graph_compiling("backward", True):
-                    compiled_bw = bw_compiler(bw_module, bw_args)
-            else:
-                fw_outs = call_func_with_args(compiled_fw, flat_tensor_args)
-            torch._C._jit_set_autocast_mode(old_jit_autocast_flag)
+    class CompiledFunction(torch.autograd.Function):
+        @staticmethod
+        @disable_torchdynamo
+        def forward(ctx, *flat_tensor_args):
+            # Disable the JIT Autocast flag to prevent re-autocasting of jitted graph.
+            # TODO - Remove when https://github.com/pytorch/functorch/pull/794 is fixed.
+            fw_outs = call_func_with_args(compiled_fw, flat_tensor_args)
             ctx.save_for_backward(*fw_outs[num_outs:])
             return tuple(fw_outs[0:num_outs])
 
         @staticmethod
         @disable_torchdynamo
         def backward(ctx, *flat_args):
-            # Disable the JIT Autocast flag to prevent re-autocasting of jitted graph.
-            # TODO - Remove when https://github.com/pytorch/functorch/pull/794 is fixed.
-            old_jit_autocast_flag = torch._C._jit_set_autocast_mode(False)
             contiguous_args = [t.contiguous() for t in flat_args]
             all_args = list(ctx.saved_tensors) + list(contiguous_args)
             ctx.maybe_clear_saved_tensors()
             out = call_func_with_args(compiled_bw, all_args, steal_args=True)
 
-            torch._C._jit_set_autocast_mode(old_jit_autocast_flag)
             return tuple(out)
 
-    return CompiledFunction
+    return CompiledFunction.apply
+
+
+def create_aot_dispatcher_function(
+    flat_fn, flat_args: List[Tensor], aot_config: AOTConfig, grad_enabled: bool
```

Comment:
Do we need to pass grad_enabled? Can we just query it inside the function?

Reply:
hmm... I think you're right. We originally needed this since we were compiling inside an autograd.Function for no reason and the autograd state is always disabled in there.